### PR TITLE
Adds a timer to UI / UX output

### PR DESCRIPTION
**Pull request by cdrage**
Adds a timer to the UI / UX:

```sh
Validation
 ✓  Validating component [5s]
 ✓  Checking component [6s]

Configuration changes
 ✓  Applying configuration [808ms]

Applying URL changes
 ✓  URL myurl2 already exists

Pushing to component nodejs-lhov of type local
 ✓  Waiting for component to start [61ms]
 ✓  Copying files to component [3s]
 ✓  Building component [6s]
 ✓  Changes successfully pushed to component
```

related to https://github.com/openshift/odo/issues/1726

### CODE REVIEW

**Comment by cdrage** [comment]
Preview:

[asciicast](https://asciinema.org/a/Hc5bsG25wG4v9cteUGfm5Fy8C)

**Comment by cdrage** [comment]
Let me know your thoughts @kadel as this implementation adds it to the non-verbose output!

**Comment by kadel** [comment]
> Closes #1726

this is not closing #1726. We need to measure this in the CI, and we also need to measure how long it takes to see changes in the running application and record thoe values so we can watch it over time.

**Comment by kadel** [comment]
/retest

```
• Failure [130.963 seconds]
odoCmpE2e
/go/src/github.com/openshift/odo/tests/integration/cmp_test.go:13
  odo component updating
  /go/src/github.com/openshift/odo/tests/integration/component.go:299
    should be able to create a git component and update it from local to git [It]
    /go/src/github.com/openshift/odo/tests/integration/component.go:310

    No future change is possible.  Bailing out early after 72.553s.
    Expected
        <int>: 1
    to match exit code:
        <int>: 0

    /go/src/github.com/openshift/odo/tests/helper/helper_run.go:29
------------------------------
```

File: pkg/log/fidget/spinner.go [hunk 1]
```diff
@@ -117,3 +119,14 @@ func (s *Spinner) Start() {
 func (s *Spinner) Stop() {
 	s.stop <- struct{}{}
 }
+
+// TimeSpent returns the seconds spent since the spinner first started
+func (s *Spinner) TimeSpent() time.Duration {
+	t := time.Now()
```
Comment (mik-dass) [inline pkg/log/fidget/spinner.go]: Maybe a better descriptive variable name than `t`

**Comment by kadel** [comment]
> ```shell
> Validation
>  ✓  Validating component [5s]
>  ✓  Checking component [6s]
> ```



A little bit unrelated question: What is the difference between `Validating component` and ` Checking component `?

**Pull request merged**